plugin/folding_range.py

```diff
@@ -0,0 +1,207 @@
+            view_change_count = self.view.change_count()
+            # If the stored change_count matches the view's actual change count, the request has already been run for
+            # this document state (i.e. "Edit" menu was opened before) and the results are still valid - no need to send
+            # another request.
+            if self.change_count == view_change_count:
+                return False
+            self.change_count = -1
+            session = self.best_session(self.capability)
+            if session:
+                params = {'textDocument': text_document_identifier(self.view)}  # type: FoldingRangeParams
+                session.send_request_async(
+                    Request.foldingRange(params, self.view),
+                    partial(self._handle_response_async, view_change_count)
+                )
+            return False
+        return self.folding_region is not None  # Already set or unset by self.description
+
+    def _handle_response_async(self, change_count: int, response: Optional[List[FoldingRange]]) -> None:
+        self.change_count = change_count
+        self.folding_ranges = response or []
+
+    def description(
+        self,
+        prefetch: bool = False,
+        hidden: bool = False,
+        strict: bool = True,
+        event: Optional[dict] = None,
+        point: Optional[int] = None
+    ) -> str:
+        if not prefetch:
+            return "LSP: Fold"
+        # Implementation detail of Sublime Text: TextCommand.description is called *before* TextCommand.is_visible
+        self.folding_region = None
+        if self.change_count != self.view.change_count():  # Ensure that the response has already arrived
+            return "LSP <debug>"  # is_visible will return False
+        if point is not None:
+            pt = point
+        else:
+            selection = self.view.sel()
+            if len(selection) != 1 or not selection[0].empty():
+                return "LSP <debug>"  # is_visible will return False
+            pt = selection[0].b
+        for folding_range in sorted_folding_ranges(self.folding_ranges):
+            region = range_to_region(folding_range_to_range(folding_range), self.view)
+            if (strict and region.contains(pt) or
+                    not strict and sublime.Region(self.view.line(region.a).a, region.b).contains(pt)) and \
+                    not self.view.is_folded(region):
+                # Store the relevant folding region, so that we don't need to do the same computation again in
+                # self.is_visible and self.run
+                self.folding_region = region
+                kind = folding_range.get('kind')
+                if kind == FoldingRangeKind.Imports:
+                    return "LSP: Fold Imports"
+                elif kind:
+                    return "LSP: Fold this {}".format(kind.title())  # pyright: ignore  # FoldingRangeKind -> StrEnum
```

Comment:
@rchl Could you add `'FoldingRangeKind': 'StrEnum',` in https://github.com/sublimelsp/lsp-python-types/blob/3b91862c0ec61c4e248e3c60cb74a836798d0a53/generate.py#L12-L15

Reply:
done

Reply:
Thanks!